From randy.dunlap@oracle.com  Wed Feb 20 14:33:28 2008
From: Randy Dunlap <randy.dunlap@oracle.com>
Date: Wed, 20 Feb 2008 13:20:50 -0800
Subject: firmware: add Kconfig and Makefile to build the firmware samples
To: Greg KH <greg@kroah.com>
Cc: akpm <akpm@linux-foundation.org>
Message-ID: <20080220132050.8f1dac05.randy.dunlap@oracle.com>


From: Randy Dunlap <randy.dunlap@oracle.com>

Now that the firmware samples are cleaned up, enable the build so that
it does not get so far behind what the kernel supports again.

Signed-off-by: Randy Dunlap <randy.dunlap@oracle.com>
Acked-by: Marcel Holtmann <marcel@holtmann.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 samples/Kconfig                 |    8 ++++++++
 samples/Makefile                |    2 +-
 samples/firmware_class/Makefile |    1 +
 3 files changed, 10 insertions(+), 1 deletion(-)

--- a/samples/Kconfig
+++ b/samples/Kconfig
@@ -33,5 +33,13 @@ config SAMPLE_KRETPROBES
 	default m
 	depends on SAMPLE_KPROBES && KRETPROBES
 
+config SAMPLE_FIRMWARE_CLASS
+	tristate "Build firmware_class examples -- loadable modules only"
+	depends on FW_LOADER && m
+	help
+	  This build firmware_class example modules.
+
+	  If in doubt, say "N" here.
+
 endif # SAMPLES
 
--- a/samples/Makefile
+++ b/samples/Makefile
@@ -1,3 +1,3 @@
 # Makefile for Linux samples code
 
-obj-$(CONFIG_SAMPLES)	+= markers/ kobject/ kprobes/
+obj-$(CONFIG_SAMPLES)	+= markers/ kobject/ kprobes/ firmware_class/
--- /dev/null
+++ b/samples/firmware_class/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_SAMPLE_FIRMWARE_CLASS) += firmware_sample_driver.o firmware_sample_firmware_class.o
